process/throttle/antiflood/quotaFloodPreventer.go

```diff
@@ -0,0 +1,113 @@
+package antiflood
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+const minMessages = 1
+const minTotalSize = 1 //1Byte
+
+type quota struct {
+	numMessages uint32
+	totalSize   uint64
+}
+
+// qoutaFloodPreventer represents a cache of quotas per peer used in antiflooding mechanism
+type quotaFloodPreventer struct {
+	mutOperation sync.RWMutex
+	cacher       storage.Cacher
+	maxMessages  uint32
+	maxSize      uint64
+}
+
+// NewQuotaFloodPreventer creates a new flood preventer based on quota / peer
+func NewQuotaFloodPreventer(
+	cacher storage.Cacher,
+	maxMessagesPerPeer uint32,
+	maxTotalSizePerPeer uint64,
+) (*quotaFloodPreventer, error) {
+
+	if cacher == nil {
+		return nil, process.ErrNilCacher
+	}
+	if maxMessagesPerPeer < minMessages {
+		return nil, fmt.Errorf("%w raised in NewCountersMap, maxMessages: provided %d, minimum %d",
```

Comment:
use constant error.

Reply:
I did use constant error as process.ErrInvalidValue but wrapped it in a new error to give more context. This is go idiomatic as of golang 1.13